src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -955,4 +997,120 @@ export class SessionSubmissionPageComponent implements OnInit, AfterViewInit {
     const userSessionEndingTime = DeadlineExtensionHelper.getOngoingUserFeedbackSessionEndingTimestamp(feedbackSession);
     return (userSessionEndingTime - Date.now()) < Milliseconds.IN_FIFTEEN_MINUTES;
   }
+
+  /**
+   * Filter questions that we are submitting for intended recipient
+   * when grouped session view is toggled and save the responses after.
+   */
+  saveResponsesForSelectedRecipientQuestions(recipientId: string,
+    questionSubmissionForms: QuestionSubmissionFormModel[]): void {
+    const questionsToRecipient: Set<number> = this.recipientQuestionMap.get(recipientId) || new Set();
```

Comment:
Just checking if it's possible for question map to return null? Similar to the other comment, if I'm not mistaken this means a bug somewhere. Shouldn't silently save here, user might think responses are saved but they aren't.

Reply:
You are right. I will change it to checking undefine and show error toast if it is undefine.